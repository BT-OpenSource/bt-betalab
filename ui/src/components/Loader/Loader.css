.loading {
    padding: 4em 0;
}