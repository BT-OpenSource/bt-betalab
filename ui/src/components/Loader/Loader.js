import React, { Component } from 'react';
import "react-table/react-table.css";
import "./Loader.css";

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

class Loader extends Component {

    render() {

        if (this.props.loading) {
            return (
                <div className="loading">
                    <FontAwesomeIcon icon="spinner" pulse size="4x" />
                </div>
            );
        } else {
            return (
                <div>
                    {this.props.children}
                </div>
            );
        }
    }
}

export default Loader;
