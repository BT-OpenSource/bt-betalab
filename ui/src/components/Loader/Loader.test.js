import React from "react";
import Loader from "./Loader";
import { shallow, mount } from "enzyme";

describe("Loader", () => {

    let shallowComponent;
    let mountedComponent;
    let shallowInstance;
    let mountedInstance;

    beforeEach(() => {
        shallowComponent = shallow(<Loader />);
        mountedComponent = mount(<Loader />);
        shallowInstance = shallowComponent.instance();
        mountedInstance = mountedComponent.instance();
    });

    describe('init()', () => {
        it('should create with no errors', () => {
            expect(shallowComponent).toMatchSnapshot();
            expect(mountedComponent).toMatchSnapshot();
        });
    });

});


