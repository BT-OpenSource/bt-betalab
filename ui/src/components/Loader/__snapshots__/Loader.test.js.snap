// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader init() should create with no errors 1`] = `<div />`;

exports[`Loader init() should create with no errors 2`] = `
<Loader>
  <div />
</Loader>
`;
